a10_octavia/controller/worker/tasks/a10_network_tasks.py

```diff
@@ -1119,10 +1122,125 @@ def execute(self, vthunder, use_device_flavor):
 class ValidateSubnet(BaseNetworkTask):
 
     def execute(self, member):
-        if member.subnet_id:
-            member_subnet = self.network_driver.get_subnet(member.subnet_id)
-            subnet_ip, subnet_mask = a10_utils.get_net_info_from_cidr(member_subnet.cidr)
-            if not a10_utils.check_ip_in_subnet_range(
-                    member.ip_address, subnet_ip, subnet_mask):
-                raise exceptions.IPAddressNotInSubnetRangeError(
-                    member.ip_address, member_subnet.cidr)
+        member_list = member if isinstance(member, list) else [member]
+        for member in member_list:
+            if member.subnet_id:
+                member_subnet = self.network_driver.get_subnet(member.subnet_id)
+                subnet_ip, subnet_mask = a10_utils.get_net_info_from_cidr(member_subnet.cidr)
+                if not a10_utils.check_ip_in_subnet_range(
+                        member.ip_address, subnet_ip, subnet_mask):
+                    raise exceptions.IPAddressNotInSubnetRangeError(
+                        member.ip_address, member_subnet.cidr)
+
+
+class DeleteSubnetAddressAndNatPoolForBatchMembers(BaseNetworkTask):
```

Comment:
I think it is better we can aggregate the reference count. And then update db, reserver/release addresses.
The total db query and address reserver/release operation is the same as before.